ironfish/src/network/peerNetwork.ts

```diff
@@ -1204,6 +1211,13 @@ export class PeerNetwork {
     peer.knownBlockHashes.set(block.header.hash, KnownBlockHashesValue.Received)
 
     // verify the block header
+    if (block.header.sequence === GENESIS_BLOCK_SEQUENCE) {
+      peer.punish(BAN_SCORE.MAX, VerificationResultReason.GOSSIPED_GENESIS_BLOCK)
```

Comment:
I think we can drop the ban here, since we don't do that for invalid blocks.

Reply:
Dropped